src/util.h

```diff
@@ -357,4 +358,14 @@ std::unique_ptr<T> MakeUnique(Args&&... args)
     return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
 }
 
+//! Simplification of std insertion
+template <typename Tdst, typename Tsrc>
+void insert(Tdst& dst, const Tsrc&& src) {
```

Comment:
As we discussed you might not need `&&` because you can pass temporary object to the function with `const Tsrc&`